packages/docker-node/src/ckb-docker-network.ts

```diff
@@ -63,6 +69,110 @@ lock.hash_type = "type"\n`
     return container.start()
   }
 
+  private async doDeploy(
+    rpc: RPC,
+    indexer: Indexer,
+    from: Address,
+    privateKey: string,
+    script: string,
+    filePath: string,
+  ): Promise<InfraScript> {
+    const scriptBinary = fs.readFileSync(filePath)
+    const txHash = await rpc.sendTransaction(
+      this.sign(
+        (
+          await commons.deploy.generateDeployWithDataTx({
+            cellProvider: indexer,
+            scriptBinary,
+            fromInfo: from,
+          })
+        ).txSkeleton,
+        privateKey,
+      ),
+    )
+
+    return {
+      name: script,
+      path: filePath,
+      cellDep: {
+        depType: 'code',
+        outPoint: {
+          txHash,
+          index: '0x0',
+        },
+      },
+    }
+  }
+
+  private async deployBuiltInScripts(
+    scriptName: string[],
+    builtInDirPath: string,
+    indexer: Indexer,
+    rpc: RPC,
+    from: Address,
+    privateKey: string,
+  ): Promise<InfraScript[]> {
+    const scripts: InfraScript[] = []
+    for (const script of scriptName) {
+      const filePath = path.join(builtInDirPath, script)
+      scripts.push(await this.doDeploy(rpc, indexer, from, privateKey, script, filePath))
+    }
+
+    return scripts
+  }
+
+  private sign(txSkeleton: helpers.TransactionSkeletonType, privateKey: string): Transaction {
+    txSkeleton = commons.common.prepareSigningEntries(txSkeleton)
+    const signature = hd.key.signRecoverable(txSkeleton.get('signingEntries').get(0)!.message, privateKey)
+    return helpers.sealTransaction(txSkeleton, [signature])
+  }
+
+  private async deployCustomScripts(
+    filePath: string,
+    indexer: Indexer,
+    rpc: RPC,
+    from: Address,
+    privateKey: string,
+  ): Promise<InfraScript[]> {
+    const scripts: InfraScript[] = []
+    if (fs.existsSync(filePath)) {
+      const configs = JSON.parse(fs.readFileSync(filePath).toString())
+      if (configs.custom && Array.isArray(configs.custom)) {
+        for (const script of configs.custom) {
+          scripts.push(await this.doDeploy(rpc, indexer, from, privateKey, script, script.path))
+        }
+      }
+    }
+
+    return scripts
+  }
```

Comment:
How about deploying built-in scripts via the workflow for custom scripts so there won't be two methods to deploy a script.
Specifically, why are `scriptName: string[]` required in `deployBuiltInScripts`

Reply:
> How about deploying built-in scripts via the workflow for custom scripts so there won't be two methods to deploy a script. Specifically, why are `scriptName: string[]` required in `deployBuiltInScripts`

I have extract the logic to the method `doDeploy`, and I think it is conveniently for other modification by separating them to two methods.

Reply:
The point I want to deliver is that `custom` and `built-in` scripts can be treated in the same manner. They're all binaries to be loaded.

Here we have two methods to load binaries, one from **built-in file path**, one from **config.custom**

What if we preset `built-in scripts` in the generated `config.builtIn`.

In fact, `config.custom` can be generalized as `config.scripts` with `built-in scripts` as its default value. Custom scripts can be appended to `config.scripts`.

But we can leave it now, because it's just about preset **explicit** or **implicit**

Reply:
Tracked by https://github.com/ckb-js/kuai/issues/304